server/src/main/java/org/opensearch/cluster/routing/allocation/AwarenessReplicaBalance.java

```diff
@@ -31,19 +33,55 @@
  * Helps in balancing shards across all awareness attributes and ensuring high availability of data.
  */
 public class AwarenessReplicaBalance {
+    public static final String SETTING_CLUSTER_ROUTING_ALLOCATION_AWARENESS_BALANCE = "cluster.routing.allocation.awareness.balance";
+    public static final String SETTING_FORCE_AWARENESS_REPLICA = "cluster.force_awareness_replica";
```

Comment:
Wondering if this could be as generic and simple as default replica count so that it could be leveraged for a wide variety of use cases even outside of awareness

Reply:
I think this is a very specific usecase that we need. For general purpose, I think we should use index_templates only